File: implementation/scipamato/scipamato-web/src/main/java/ch/difty/scipamato/web/autoconfiguration/BootstrapProperties.java
package ch.difty.scipamato.web.autoconfiguration;

import org.springframework.boot.context.properties.ConfigurationProperties;

import de.agilecoders.wicket.core.settings.BootstrapSettings;
import de.agilecoders.wicket.themes.markup.html.bootswatch.BootswatchTheme;

@ConfigurationProperties(prefix = BootstrapProperties.PROPERTY_PREFIX)
public class BootstrapProperties extends BootstrapSettings {

    public static final String PROPERTY_PREFIX = "wicket.external.agilcoders.bootstrap";

    private boolean enabled = true;

    private BootswatchTheme theme = BootswatchTheme.Sandstone;

    public BootswatchTheme getTheme() {
        return theme;
    }

    public void setTheme(BootswatchTheme theme) {
        this.theme = theme;
    }

    public boolean isEnabled() {
        return enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }
}

File: public/public-web/src/main/kotlin/ch/difty/scipamato/publ/web/autoconfiguration/BootstrapProperties.kt
package ch.difty.scipamato.publ.web.autoconfiguration

import de.agilecoders.wicket.core.settings.BootstrapSettings
import de.agilecoders.wicket.themes.markup.html.bootswatch.BootswatchTheme
import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = BootstrapProperties.PROPERTY_PREFIX)
data class BootstrapProperties(
    val isEnabled: Boolean = true,
    val theme: BootswatchTheme = BootswatchTheme.Sandstone,
) : BootstrapSettings() {

    companion object {
        const val PROPERTY_PREFIX = "wicket.external.agilecoders.bootstrap"
    }
}
